### PR TITLE
AISOTT: Minor mapping fixes and new index on listen table

### PR DESCRIPTION
**Pull request by mayhem**
This PR fixes a couple of minor issues that were not right with the latest mapping refactoring:

1. Fix the incoming listens match rate
2. Add an index on recording_msid (JSONB) on listens -- this becomes more important as we do more re-checks.

### CODE REVIEW

File: admin/timescale/create_indexes.sql [hunk 1]
```diff
@@ -3,6 +3,7 @@ BEGIN;
 CREATE INDEX listened_at_user_name_ndx_listen ON listen (listened_at DESC, user_name);
 CREATE INDEX created_ndx_listen ON listen (created);
 CREATE UNIQUE INDEX listened_at_track_name_user_name_ndx_listen ON listen (listened_at DESC, track_name, user_name);
+CREATE INDEX recording_msid_ndx_listen on listen ((data->'track_metadata'->'additional_info'->>'recording_msid'));
```
Comment (amCap1712) [inline admin/timescale/create_indexes.sql:6]: Assuming this is for https://github.com/metabrainz/listenbrainz-server/blob/3f24a5309480c52c9c872c63a56d18419ce9c56c/listenbrainz/mbid_mapping_writer/job_queue.py#L153 

Adding this index is probably fine but if postgres decides to do a hash join, the index probably won't improve performance. So may want to check execution plans before and after creating the index.

File: listenbrainz/mbid_mapping_writer/job_queue.py [hunk 2]
```diff
@@ -260,15 +260,17 @@ def update_metrics(self, stats):
                         low_quality_rate=stats["low_quality"] - stats["last_low_quality"],
                         no_match_rate=stats["no_match"] - stats["last_no_match"],
                         listens_per_sec=listens_per_sec,
-                        listen_count=stats["listen_count"],
-                        listens_matched=stats["listens_matched"],
+                        listens_matched_p=stats["listens_matched"] / stats["listen_count"] * 100.0,
```
Comment (amCap1712) [inline listenbrainz/mbid_mapping_writer/job_queue.py:263]: Calculation looks good but don't know what format etc grafana expects this to be in.

File: admin/timescale/create_indexes.sql (hunk 1 above)
Comment (mayhem) [reply admin/timescale/create_indexes.sql:6]: This index is less for the current PR and more for the other things I often do in working with the mapping. For instance, in order to select MSIDs to recheck, I need to query on recording_msid frequently and its slow without an index. Much faster with one -- and in some cases the queries become near instant.

File: listenbrainz/mbid_mapping_writer/job_queue.py (hunk 2 above)
Comment (mayhem) [reply listenbrainz/mbid_mapping_writer/job_queue.py]: As per our current discussions in IRC, I find it easier to do the calculations in python than wrestling with the influx query syntax, so calculating what I want to show (%) is just easier.

**Pull request closed without merging**